test/test-projects/junit/src/test/java/junit5/ParallelExecutionTest.java

```diff
@@ -0,0 +1,46 @@
+package junit5;
+
+import java.util.concurrent.Semaphore;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.parallel.Execution;
+import org.junit.jupiter.api.parallel.ExecutionMode;
+
+/**
+ * Test class with 2 test methods which are executed concurrently, so that the
+ * correct result processing of overlapping test runs can be verified.
+ */
+@Execution(ExecutionMode.CONCURRENT)
+class ParallelExecutionTest {
+
+    // make sure that both test executions overlap,
+    // thus both reporting their starts before any reports its end
+    private static final Semaphore s1 = new Semaphore(0);
+    private static final Semaphore s2 = new Semaphore(0);
+
+    @Test
+    void test1() {
+        s1.release();
+        s2.acquireUninterruptibly();
+        // fail("test1");
```

Comment:
Is the code here be commented out on purpose?

Reply:
not really - I removed the commented-out code and updated the class so that it now includes two failing tests as well as a passing one